tools/py/committee-reweight.py

```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/env python3
+import sys
+import argparse
+import numpy as np
+from ipi.inputs.simulation import InputSimulation
+from ipi.utils.io.inputs import io_xml
+
+
+def direct_reweight(pot, obs, kbT):
+    """
+    reweights an observable based on a committee of potentials, at the temperature kT
+
+    pot: an array (nframes x ncommittee) containing the values of the potentials for each committee
+    obs: an array containing the values of the observables
+    kbT: the temperature, in energy units
+    """
+    beta = 1.0 / kbT
+    num_pot_frames = pot.shape[0]
+    num_obs_frames = obs.shape[0]
+    if num_pot_frames != num_obs_frames:
+        raise RuntimeError(
+            "Potential and observable files have different numbers of frames"
+        )
+
+    num_pot_models = pot.shape[1]
+    if obs.ndim == 1:
+        obs = np.expand_dims(obs, axis=1)
+
+    # vectorized evaluation of weights
+    weights = np.array(-beta * (pot.T - np.mean(pot, axis=1)).T, dtype=np.float128)
+    weights = np.exp(weights)
+
+    # 1-d array of lenght num_pot_models: normalization of the weights over frames
+    norm = np.sum(weights, axis=0)
+    for i in range(num_pot_models):
+        weights[:, i] /= norm[i]
+    obs_avg_rew = obs.T @ weights
+
+    return obs_avg_rew, weights
+
+
+def CEA(pot, obs, kbT):
+    """
+    reweight the quantity by cumulant expansion approximation (CEA)
+    ref:
+    Michele Ceriotti,  Guy A. R. Brain, Oliver Riordan and David E.
+    Manolopoulos 2011The inefficiency of re-weighted sampling and
+    the curse of system size in high-order path integration
+    Proc. R. Soc. A.4682–17
+    http://doi.org/10.1098/rspa.2011.0413
+    :return:
```

Comment:
missing? return what?

Reply:
Added proper indication of the inputs and outputs of all of the functions. Tried to keep a uniform style.